### PR TITLE
#157995248 Implement Feedback

### PR DESCRIPTION
**Pull request by asamisellah**
#### What does this PR do?
Create CRUD API Endpoints 

#### Description of tasks to be completed?
METHOD | ENDPOINT | FUNCTIONALITY 
------------ | -------------|--------------
GET | /api/v1/users/requests/ | Fetch all the requests resources
GET | /api/v1/users/requests/requestId | Fetch a single request resource
PUT | /api/v1/users/requests/requestId | Modify a request
POST | /api/v1/users/requests | Create a new request
DELETE | /api/v1/users/requests/requestId | Delete a specific request
GET | /api/v1/users/signup/ | Handle user registration
POST | /api/v1/users/signin/ | Handle user login

#### How should this be manually tested?
1. clone the [repo](https://github.com/asamisellah/Maintenance-Tracker)
2. open the folder in your text editor or IDE
3. `git checkout -b new-branch`
4. `git fetch ch-env-setup-157930685`

*Setup*
Make sure virtual environment is installed. Open terminal and run the following commands
1. `cd maintenance-tracker`
2.  `virtualenv venv` to create virtual environment
3.  `source venv/bin/activate` to run environment
4. `pip install -r requirements.txt`

*Testing*
run ` pytest`

#### Any background context you want to provide?
None

#### What are the relevant pivotal tracker stories?
[#157965217 ](https://www.pivotaltracker.com/story/show/157965217/comments/190109380)
[#157995248](https://www.pivotaltracker.com/story/show/157995248/comments/190191728)

### Screenshot
signin_user
Create User

create_user
Sign In user

post_request
Create Request

get_requests
Get  #Requests

### Questions
Have I satisfied feedback
Are the endpoints functioning?

### CODE REVIEW

File: app/__init__.py [hunk 1]
```diff
@@ -3,69 +3,72 @@
...
 # POST a user
-@app.route('/users', methods=['POST'])
+@app.route('/api/v1/users', methods=['POST'])
 def create_user():
     user = {
         "username": request.json.get("username"),
         "email": request.json.get('email'),
         "password": request.json.get("password"),
-        "cconfirm_password": request.json.get('cconfirm_password'),
+        "confirm_password": request.json.get('confirm_password')
```
Comment (zmwas) [inline app/__init__.py]: Confirm password should not be in the user object, the id should be in the object. Your create user function should have a way to check if the user with the username or email already exists before it is appended. Write tests to check those edge cases.

File: test_users.py [hunk 2]
```diff
@@ -1,49 +1,32 @@
...
     def setUp(self):
         self.client = app.test_client()
 
-    def register_user(self, name, email, password, confirm_password):
-        
-        return self.client.post('/register', data=self.users)
-
-    def test_registration(self):
-        res = self.client.post('/register', data=self.users)
+    def test_create_user(self):
```
Comment (zmwas) [inline test_users.py]: Add edge case to test for similar usernames, bad emails e.t.c
